Start draft with image block

mzorz opened the pull request:
This PR brings in the functionality to:

a) start a Post by sharing an image to WordPress app from the outside, and
b) start a Post by inserting an image to WordPress from the Media section's "upload finished" snackbar.

Please note the bridge and changes on Android were implemented, but this needs an iOS dev to take care of the iOS part.

To test:

Follow the description in the WPAndroid PR https://github.com/wordpress-mobile/WordPress-Android/pull/9442.

**Note:** First make sure to have the `Use Block Editor` setting to `ON` in App Settings. Then:

#### CASE A: share from another app

1. open the Photos app
2. pick a picture and tap the share icon
3. select the WordPress app
4. choose a site and tap on `Add to new post`
5. Observe the Gutenberg editor opens up, with a pre-filled `Image Block` and the selected image starts uploading.

![share_media_from_outside](https://user-images.githubusercontent.com/6597771/54575567-4e2cd980-49d3-11e9-8ea6-e95c7ecdd62c.gif)




####  CASE B: start draft from Media Section
1. open the app
2. go to the Media section
3. tap on the `+` icon on the right-top corner of the screen
4. select one option to upload an image to the Media section
5. wait for it to finish uploading
6. once it's uploaded, observe a snackbar appears, with action button `WRITE POST`. Tap on it.
7. observe the Gutenberg editor is fired, with a pre-filled `Image Block` and the selected image starts uploading.

![share_media_from_media_section_snackbar2](https://user-images.githubusercontent.com/6597771/54575813-5c2f2a00-49d4-11e9-8883-d035454c0121.gif)

## EDIT: added test cases for Multiple images

Commits 8e84d41 and 3cc1114 contain changes that allow for multiple images to be shared into a Post, adding the test cases here:

#### CASE C: editor not available right away
1. kill the app
2. go to the Photos app
3. select several photos
4. tap share, select WordPress app
5. observe the Gutenberg editor is fired up, and several image blocks are appended, with the images on each one of them.
6. all should upload successfully, so wait for it to upload
7. also test exiting the editor while things are uploading, etc.

#### CASE D: editor available right away
1. after running CASE C, the app will be available
2. go ahead and repeat the action, this time the code path will be slightly different, but the user experience should be the same: repeat steps 2-7 of CASE C and verify it all works normally.


#### CASE E: multiple shared images uploading, then adding a new image block and picking file from device.
1. go to the Photos app
2. select several photos
3. tap share, select WordPress app
4. observe the Gutenberg editor is fired up, and several image blocks are appended, with the images on each one of them.
5. while this is happening, use the inserter and insert yet another Image block
6. choose a different image (i.e. take a picture)
7. observe all of them get uploaded correctly (even exit the editor and observe they're all uploaded async)
8. open the Editor again and confirm it's all there as expected.

marecar3 commented on react-native-gutenberg-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/WPAndroidGlueCode.java:
```diff
@@ -48,9 +48,12 @@
     private OnMediaLibraryButtonListener mOnMediaLibraryButtonListener;
     private OnReattachQueryListener mOnReattachQueryListener;
     private OnEditorMountListener mOnEditorMountListener;
+    private boolean mEditorIsMounted = false;
```
It would be maybe cleaner to name it `isEditorMounted` as it's boolean.
and not sure if we need to set it as false, as Java does it automatically for us?

mzorz replied:
good! 👌 addressed in 4f744ccc

mzorz commented:
Hi @marecar3 ! I just added commits 8e84d41 and 3cc1114, these contain changes that allow for multiple images to be shared into a Post, and updated the description of the PR to include new test cases (cases C, D, and E). Could you give this PR another round if possible? So we make sure this works correctly as well :) - thanks in advance!

marecar3 commented:
Hey @mzorz, not a problem, I will do it right away! :)

marecar3 commented:
LGTM!!! @mzorz ! Multiple photos scenario is working like expected :)
